include/micm/process/cuda_process_set.hpp

```diff
@@ -31,11 +31,10 @@ namespace micm
     void SetJacobianFlatIds(const SparseMatrix<double, OrderingPolicy>& matrix);
 
     template<template<class> typename MatrixPolicy>
-    requires VectorizableDense<MatrixPolicy<double>> std::chrono::nanoseconds AddForcingTerms(
+    requires VectorizableDense<MatrixPolicy<double>> void AddForcingTerms(
         const MatrixPolicy<double>& rate_constants,
         const MatrixPolicy<double>& state_variables,
-        MatrixPolicy<double>& forcing)
-    const;
+        MatrixPolicy<double>& forcing);
```

Comment:
Why can this no longer be a `const` function?

Reply:
Thanks @mattldawson . I modified `this->devstruct` inside the `AddForcingTerms` function now (https://github.com/NCAR/micm/blob/update_cuda_class_for_addforcing/include/micm/process/cuda_process_set.hpp#L98-L100) and I thought I could not make it `const` anymore. If it is not the case or we can do it in the constructor instead, I am happy to put `const` back.

Reply:
Ah, ok - I would probably set these values in the constructor. If you want, you can do an assert in this function that they are the correct values, but having these functions be `const` is how we ensure thread-safety for the `Rosenbrock::Solve()` function

Reply:
Or you could also create a copy of `this->devstruct_` and modify that, but since `ProcessSet` is created for and only works with matrices of specific sizes, my vote would be to do it in the constructor.

Reply:
Thanks @mattldawson and @mwaxmonsky for your comments. I have updated the implementations to use the `CudaVectorMatrixParams` struct to convey those integer variables instead. Let me know if the new implementation looks better.